File: alligator/src/main/java/me/aartikov/alligator/animations/SimpleTransitionAnimation.java
package me.aartikov.alligator.animations;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.AnimRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

/**
 * Date: 26.03.2017
 * Time: 12:02
 *
 * @author Artur Artikov
 */

/**
 * Transition animation that uses anim resources.
 */
public class SimpleTransitionAnimation implements TransitionAnimation {
	private int mEnterAnimation;
	private int mExitAnimation;

	/**
	 * @param enterAnimation animation resource that will be used for an appearing activity/fragment
	 * @param exitAnimation  animation resource that will be used for a disappearing activity/fragment
	 */
	public SimpleTransitionAnimation(@AnimRes int enterAnimation, @AnimRes int exitAnimation) {
		mEnterAnimation = enterAnimation;
		mExitAnimation = exitAnimation;
	}

	public int getEnterAnimation() {
		return mEnterAnimation;
	}

	public int getExitAnimation() {
		return mExitAnimation;
	}

	@Override
	@Nullable
	public Bundle getActivityOptionsBundle(@NonNull Activity activity) {
		return null;
	}

	@Override
	public boolean needDelayActivityFinish() {
		return false;
	}

	@Override
	public void applyBeforeActivityStarted(@NonNull Activity currentActivity, @NonNull Intent intent) {
	}

	@Override
	public void applyAfterActivityStarted(@NonNull Activity currentActivity) {
		currentActivity.overridePendingTransition(mEnterAnimation, mExitAnimation);
	}

	@Override
	public void applyBeforeActivityFinished(@NonNull Activity activity) {
	}

	@Override
	public void applyAfterActivityFinished(@NonNull Activity activity) {
		activity.overridePendingTransition(mEnterAnimation, mExitAnimation);
	}

	@Override
	public void applyBeforeFragmentTransactionExecuted(@NonNull FragmentTransaction transaction, @NonNull Fragment enteringFragment, @NonNull Fragment exitingFragment) {
		transaction.setCustomAnimations(mEnterAnimation, mExitAnimation);
	}

	@Override
	public void applyAfterFragmentTransactionExecuted(@NonNull Fragment enteringFragment, @NonNull Fragment exitingFragment) {
	}
}

File: alligator/src/main/java/me/aartikov/alligator/animations/SimpleTransitionAnimation.kt
package me.aartikov.alligator.animations

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.annotation.AnimRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction

/**
 * Transition animation that uses anim resources.
 *
 * @param enterAnimation animation resource that will be used for an appearing activity/fragment
 * @param exitAnimation  animation resource that will be used for a disappearing activity/fragment
 */
class SimpleTransitionAnimation(
    @param:AnimRes val enterAnimation: Int,
    @param:AnimRes val exitAnimation: Int
) : TransitionAnimation {

    override fun getActivityOptionsBundle(activity: Activity): Bundle? {
        return null
    }

    override fun needDelayActivityFinish(): Boolean {
        return false
    }

    override fun applyBeforeActivityStarted(currentActivity: Activity, intent: Intent) {}
    override fun applyAfterActivityStarted(currentActivity: Activity) {
        currentActivity.overridePendingTransition(enterAnimation, exitAnimation)
    }

    override fun applyBeforeActivityFinished(activity: Activity) {}
    override fun applyAfterActivityFinished(activity: Activity) {
        activity.overridePendingTransition(enterAnimation, exitAnimation)
    }

    override fun applyBeforeFragmentTransactionExecuted(
        transaction: FragmentTransaction,
        enteringFragment: Fragment,
        exitingFragment: Fragment
    ) {
        transaction.setCustomAnimations(enterAnimation, exitAnimation)
    }

    override fun applyAfterFragmentTransactionExecuted(
        enteringFragment: Fragment,
        exitingFragment: Fragment
    ) {
    }
}